v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/endtoend/EndToEndTestingITBase.java

```diff
@@ -256,28 +432,60 @@ public PipelineLauncher.LaunchInfo launchFwdDataflowJob(
             .setPrivateConnectivity("datastream-private-connect-us-central1")
             .build();
     Stream stream =
-        createDataStreamResources(artifactBucket, gcsPrefix, jdbcSource, datastreamResourceManager);
+        createDataStreamResources(
```

Comment:
As my previous comment, the Datastream resources and the stream needs to be created before the Bulk migration job starts.